python/dsbox/controller/controller.py

```diff
@@ -359,25 +366,40 @@ def search_template(self, template: DSBoxTemplate, candidate: typing.Dict=None,
 
         # search = TemplateDimensionalSearch(template, space, d3m.index.search(), self.dataset,
         # self.dataset, metrics)
-        if self.test_dataset is None:
-            search = TemplateDimensionalSearch(
-                template, space, d3m.index.search(), self.problem_doc_metadata, self.dataset,
-                self.dataset, metrics, output_directory=self.output_directory,
-                log_dir=self.output_logs_dir, num_workers=self.num_cpus)
+
+        # split the dataset first time
+        self.train_dataset1, self.test_dataset1 = split_dataset(dataset = self.all_dataset, 
```

Comment:
Need to split dataset earlier, say in initialization. If done within search_template(), then dataset will be resplit many times.